lib/puppet/util/package.rb

```diff
@@ -1,8 +1,13 @@
 module Puppet::Util::Package
   def versioncmp(version_a, version_b)
+    
+    # remove unnecesary ending .0 or .00.0 from version
+    trimmed_version_a = version_a.sub(/(\.0+)+$/, '')
```

Comment:
```suggestion
    trimmed_version_a = version_a.sub(/(-0+|\.0+)+$/, '')
```
I think this would be good to have in case some versions have the `-` character.